packaging/opensuse/snapd.spec

```diff
@@ -171,6 +167,28 @@ pushd %{indigo_srcdir}
 # Add patch0 -p1 ... as appropriate here.
 popd
 
+# Generate snapd.defines.mk, this file is included by snapd.mk. It contains a
+# number of variable definitions that are set based on their RPM equivalents.
+# Since we can apply any conditional overrides here in the spec file we can
+# maintain one consistent set of variables across the spec and makefile worlds.
+cat >snapd.defines.mk <<__DEFINES__
+# This file is generated by openSUSE's snapd.spec
+# Directory variables.
+prefix = %{_prefix}
+bindir = %{_bindir}
+sbindir = %{_sbindir}
+libexecdir = %{_libexecdir}
+mandir = %{_mandir}
+datadir = %{_datadir}
+localstatedir = %{_localstatedir}
+unitdir = %{_unitdir}
+# Build configuration
+with_core_bits = 0
+with_alt_snap_mount_dir = %{!?with_alt_snap_mount_dir:0}%{?with_alt_snap_mount_dir:1}
+with_apparmor = %{!?with_apparmor:0}%{?with_apparmor:1}
+with_test_keys = %{!?with_test_keys:0}%{?with_test_keys:1}
+__DEFINES__
```

Comment:
Wouldn't this be easier if we had something like a configure script where we could pass these options in and this file would be generated?

I don't have anything against this structure, other than it would make it hard to do with packaging systems like Debian's.

Reply:
I don't see how putting this in a separate script helps (it would defeat the purpose of the makefile and would focus us to have hacks for rpm defines). I also don't see any issues with generating this from `debian/rules`. I expect most systems to simply generate a fixed `snapd.defines.mk`.

Reply:
That's fair. I could equally have this as an extra source file. This is fine as-is.